pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctAggregationFunction.java

```diff
@@ -104,23 +104,23 @@ public AggregationResultHolder createAggregationResultHolder() {
 
   @Override
   public void aggregate(int length, AggregationResultHolder aggregationResultHolder,
-      Map<String, BlockValSet> blockValSetMap) {
-    int numColumns = _columns.length;
+      Map<TransformExpressionTree, BlockValSet> blockValSetMap) {
     int numBlockValSets = blockValSetMap.size();
-    Preconditions.checkState(numBlockValSets == numColumns, "Size mismatch: numBlockValSets = %s, numColumns = %s",
-        numBlockValSets, numColumns);
-
-    DistinctTable distinctTable = aggregationResultHolder.getResult();
-    BlockValSet[] blockValSets = new BlockValSet[numColumns];
-
-    for (int i = 0; i < numColumns; i++) {
-      blockValSets[i] = blockValSetMap.get(_columns[i]);
+    int numExpressions = _inputExpressions.size();
+    Preconditions
+        .checkState(numBlockValSets == numExpressions, "Size mismatch: numBlockValSets = %s, numExpressions = %s",
+            numBlockValSets, numExpressions);
+
+    BlockValSet[] blockValSets = new BlockValSet[numExpressions];
+    for (int i = 0; i < numExpressions; i++) {
+      blockValSets[i] = blockValSetMap.get(_inputExpressions.get(i));
```

Comment:
Do we agree that there will be some (may be marginal only) overhead in terms of computing equals() on TransformExpressionTree (which will compare the expression type and the expression value which is column name for the general case)? Earlier it was being done directly on String identifier. Same goes for hashcode.

Reply:
For column (not function expression), there might be minimal overhead (overhead should be much smaller comparing to creating the map). For function expression, IMO comparing TransformExpressionTree should be cheaper comparing to the expression string.
Also, here we saved the overhead of converting expression to string, so directly using expression should give better performance.